packages/expo-sensors/ios/PedometerModule.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023-present 650 Industries. All rights reserved.
+
+import CoreMotion
+import ExpoModulesCore
+
+private let EVENT_PEDOMETER_UPDATE = "Exponent.pedometerUpdate"
+
+public final class PedometerModule: Module {
+  private lazy var pedometer = CMPedometer()
+
+  private var watchStartDate: Date?
+  private var watchHandler: CMPedometerHandler?
+
+  public func definition() -> ModuleDefinition {
+    Name("ExponentPedometer")
+
+    Events(EVENT_PEDOMETER_UPDATE)
+
+    AsyncFunction("isAvailableAsync") {
+      return CMPedometer.isStepCountingAvailable()
+    }
+
+    AsyncFunction("getStepCountAsync") { (startTime: Double, endTime: Double, promise: Promise) in
+      let startDate = Date(timeIntervalSince1970: startTime / 1000.0)
+      let endDate = Date(timeIntervalSince1970: endTime / 1000.0)
+
+      pedometer.queryPedometerData(from: startDate, to: endDate) { data, error in
+        if let error {
+          promise.reject(PedometerQueryException().causedBy(error))
+        } else {
+          promise.resolve([
+            "steps": data?.numberOfSteps
```

Comment:
I think we should returns 0 if data isn't available. What do you think?

Reply:
Afaik this will never happen. The `data` can only be nil when the `error` is not nil, so the promise would be rejected instead. I can add that anyway, for clarity.

```suggestion
            "steps": data?.numberOfSteps ?? 0
```